0015-ilp-addresses/0015-ilp-addresses.md

```diff
@@ -0,0 +1,123 @@
+
+The general allocation scheme for ILP Addresses is currently the only supported scheme. It uses the prefix `g.`.
+
+This scheme has no central issuing authority or mechanism, so more than one entity can use the same address. In this case, some connectors may prepare a route to a different account than intended. In this failure case, no money moves because the receiver does not send the fulfillment. Participants in the interledger can reduce the chances of encountering this failure case by choosing addresses carefully and by properly managing connectors' routing tables.
+
+To make routing work well, we recommend including the following components as segments in an address (in order):
+
+- [Neighborhoods](#neighborhoods)
+- [Payment rail](#payment-rail)
+- [Ledger prefix](#ledger-prefix)
+- [Subledger and account group](#subledger-and-account-group)
+- [Account identifier](#account-identifier)
+- [Interactions](#interactions)
+
+
+### Neighborhoods
+
+Neighborhoods have no specific meaning, but serve as a quick shortcut for routing to the right area.
+
+- For regional currencies, use a continent code from the folowing table:
+    | Segment | Continent     |
+    |:--------|:--------------|
+    | `af`    | Africa        |
+    | `as`    | Asia          |
+    | `eu`    | Europe        |
+    | `na`    | North America |
+    | `sa`    | South America |
+    | `oc`    | Oceania       |
+    | `an`    | Antarctica    |
+- For decentralized crypto-currency ledgers, use the `crypto` neighborhood.
+- For naturally-ocurring commodities or other globally-distributed currencies, use the `earth` neighborhood.
+- For currencies strongly associated with a particular international corporation, use the continent of that corporation's headquarters.
+
+It may also be useful to have 1-3 levels of "sub-neighborhoods" to group together related connectors or ledgers. We have no specific recommendations at this time.
+
+### Payment Rail
+
+The payment rail refers to a specific way some accounts or ledgers are connected outside of ILP, such as ACH in the U.S., SEPA in Europe, or other online payment systems such as PayPal or Alipay. We recommend using an all-lowercase abbreviation for the payment rail if one is applicable to the address. Examples:
+
+| Segment | Payment Rail                                                       |
+|:--------|:-------------------------------------------------------------------|
+| `ach`   | [United States Automated Clearing House](https://en.wikipedia.org/wiki/Automated_Clearing_House) |
+| `sepa`  | [Single Euro Payments Area](https://en.wikipedia.org/wiki/Single_Euro_Payments_Area) |
+
+***TODO: more rails?***
+
+### Ledger Prefix
+
+The ledger prefix is the unique name of the ledger. When doing a local delivery, the connector typically uses this segment to figure out which ledger plugin to use. As a result, it's difficult (but possible) to make a connector between two ledgers that use the same ledger prefix.
```

Comment:
I see 3 main options:

- Ledger chooses - the operator of a ledger, such as an instance of the Five Bells Ledger, can choose a ledger prefix and have the ledger report that prefix, for example through the API
- Plugin chooses - in the case of a cryptocurrency, the cryptocurrency itself wouldn't "tell you" what it's prefix is. But it would be in the interest of anyone writing a ledger plugin for that ledger to use the same value and hardcode it in their implementation. So when a client or connector asks the plugin for the prefix, it will return that value.
- Connectors choose - in case neither of the previous options are possible, connectors to a given ledger may choose to settle upon a common identifier for that ledger